doc/api/test.md

```diff
@@ -317,6 +317,23 @@ In this example, the test runner will execute the files `test1.js` and
 `test2.mjs`. The test runner will also recursively search the
 `custom_test_dir/` directory for test files to execute.
 
+## Watch mode
```

Comment:
Can you move this section. Otherwise, it will look like `### Test runner execution model` is part of watch mode due to the headings.